tests/util_tests/test_version_compare.py

```diff
@@ -10,62 +10,103 @@
 
 from tank_test.tank_test_base import setUpModule  # noqa
 from tank_test.tank_test_base import ShotgunTestBase
+from sgtk.util import (
+    is_version_older,
+    is_version_newer,
+    is_version_newer_or_equal,
+    is_version_older_or_equal,
+)
+
+OLDER = "older"
+NEWER = "newer"
+EQUAL = "equal"
+
+git_sha = "b2cbcb9cefea668eb4ccf071e51cc650ebb27504"
 
 
 class TestVersionCompare(ShotgunTestBase):
     def setUp(self):
         super(TestVersionCompare, self).setUp()
 
-    def test_is_version_newer(self):
-
-        from tank.util.version import is_version_newer
-
-        self.assertTrue(is_version_newer("1.2.3", "1.0.0"))
-        self.assertTrue(is_version_newer("v1.2.3", "v1.0.0"))
-        self.assertTrue(is_version_newer("v1.2.3", "1.0.0"))
-
-        self.assertTrue(is_version_newer("v1.200.3", "v1.12.345"))
-
-        self.assertTrue(is_version_newer("HEaD", "v1.12.345"))
-        self.assertTrue(is_version_newer("MAsTER", "v1.12.345"))
-
-        self.assertTrue(is_version_newer("HEaD", "1.12.345"))
-        self.assertTrue(is_version_newer("MAsTER", "1.12.345"))
-
-        self.assertFalse(is_version_newer("v0.12.3", "HEaD"))
-        self.assertFalse(is_version_newer("v0.12.3", "MAsTER"))
-
-        self.assertFalse(is_version_newer("0.12.3", "HEaD"))
-        self.assertFalse(is_version_newer("0.12.3", "MAsTER"))
-
-        self.assertFalse(is_version_newer("v0.12.3", "1.2.3"))
-        self.assertFalse(is_version_newer("v0.12.3", "0.12.4"))
-
-        self.assertFalse(is_version_newer("1.0.0", "1.0.0"))
-
-    def test_is_version_older(self):
-
-        from tank.util.version import is_version_older
-
-        self.assertFalse(is_version_older("1.2.3", "1.0.0"))
-        self.assertFalse(is_version_older("v1.2.3", "v1.0.0"))
-        self.assertFalse(is_version_older("v1.2.3", "1.0.0"))
-
-        self.assertFalse(is_version_older("v1.200.3", "v1.12.345"))
-
-        self.assertFalse(is_version_older("HEaD", "v1.12.345"))
-        self.assertFalse(is_version_older("MAsTER", "v1.12.345"))
-
-        self.assertFalse(is_version_older("HEaD", "1.12.345"))
-        self.assertFalse(is_version_older("MAsTER", "1.12.345"))
-
-        self.assertTrue(is_version_older("v0.12.3", "HEaD"))
-        self.assertTrue(is_version_older("v0.12.3", "MAsTER"))
-
-        self.assertTrue(is_version_older("0.12.3", "HEaD"))
-        self.assertTrue(is_version_older("0.12.3", "MAsTER"))
-
-        self.assertTrue(is_version_older("v0.12.3", "1.2.3"))
-        self.assertTrue(is_version_older("v0.12.3", "0.12.4"))
-
-        self.assertFalse(is_version_older("1.0.0", "1.0.0"))
+    versions = {
+        ("1.2.3", "1.2.3"): EQUAL,
```

Comment:
I combed through the list of values we tested for in both tests and these are all the cases, with the appropriate result.